diff --git a/js/ui/keyboard.js b/js/ui/keyboard.js
index 1666a507b86e99d52dac0c9d30ded41a871a075e..4a249404450f8cd7c7f894ef3427bc5201b94f3b 100644
--- a/js/ui/keyboard.js
+++ b/js/ui/keyboard.js
@@ -526,9 +526,7 @@ var FocusTracker = new Lang.Class({
         this._currentWindow = window;
         if (window) {
             this._currentWindowPositionId = this._currentWindow.connect('position-changed', () => {
-                if (global.display.get_grab_op() == Meta.GrabOp.NONE)
-                    this.emit('position-changed');
-                else
+                if (global.display.get_grab_op() != Meta.GrabOp.NONE)
                     this.emit('reset');
             });
         }

