diff --git a/src/libtracker-data/meson.build b/src/libtracker-data/meson.build
index e92fa800d7b7c32e3c1cf8da9826ecaf42839970..fa1ebcbdbed3f8afa289de916bb00cecc826bc9c 100644
--- a/src/libtracker-data/meson.build
+++ b/src/libtracker-data/meson.build
@@ -73,7 +73,7 @@ libtracker_data = library('tracker-data',
     # symbols from libtracker-sparql, but does not mean it should
     # be linked to it.
     override_options: ['b_lundef=false'],
-    dependencies: tracker_data_dependencies + [unicode_library],
+    dependencies: tracker_data_dependencies + [unicode_library, tracker_sparql_intermediate_dep],
     include_directories: [commoninc, configinc, srcinc],
 )
 
diff --git a/src/libtracker-fts/meson.build b/src/libtracker-fts/meson.build
index ca892cf0132535eb52c40bc8efe5f50ff17d280d..cbfe553a14a10d90ba3f06764e981ae5253ebbef 100644
--- a/src/libtracker-fts/meson.build
+++ b/src/libtracker-fts/meson.build
@@ -11,7 +11,7 @@ libtracker_fts = static_library('tracker-fts',
     'tracker-fts-config.c',
     'tracker-fts-tokenizer.c',
     libtracker_fts_fts5,
-    dependencies: [tracker_common_dep],
+    dependencies: [tracker_common_dep, tracker_sparql_intermediate_dep],
     c_args: tracker_c_args
 )
 

