--- a/src/backends/x11/meta-renderer-x11.c	2017-10-30 23:24:49.000000000 +0100
+++ b/src/backends/x11/meta-renderer-x11.c	2018-01-18 14:36:49.297637981 +0100
@@ -80,7 +80,7 @@
    * to work properly, even when Mutter is constantly redrawing new frames;
    * otherwise, without INTEL_swap_event, we'll just block in glXSwapBuffers().
    */
-  cogl_xlib_renderer_set_threaded_swap_wait_enabled (cogl_renderer, TRUE);
+  cogl_xlib_renderer_set_threaded_swap_wait_enabled (cogl_renderer, FALSE);
 
   return cogl_renderer;
 }
